### DISCUSSION TITLE
Add a Whitelist/Blacklist tag field for containers in Submarine Editor to bypass/restrict valid items

### DISCUSSION THREAD

**Discussion by Svenn-20** [FakeFishGames/Barotrauma#15152] [Ideas and suggestions]
To clarify, I'm not talking about the "Container Restrictions" tag field which only governs AI cleanup and Autofill.
I think a tag field that allows you to customize what items are valid to be placed in a container would have a little bit of utility for weapon-system cosmetics, and have no effect on your average player.

I propose being able to change the valid items of a container through whitelists/blacklists so that, for example, a railgun loader will accept depth charges, and a railgun will be able to use them as ammo.

Ammo Loaders of every type are currently restricted to only accept items tagged with a matching ammo type, and can't be customized to just accept multiple ammo types.
Specific example: Changing the tags on boarding pods to be railgun ammo would let them be fired from railguns of course, but **_obtained_** boarding pods would have the default tags and can't be fired in the same way, since the loaders are restricted to only take "railgunammo" tagged items, and adding "depthchargeammosource" to the railgun loader still doesn't allow depth charges to be placed inside.

I'm sure there are already methods of doing what I'm trying to do through modding, but a whitelist/blacklist field would make my project much simpler through just the submarine editor AND have potential outside of just weapon systems, like custom specialized crates or buckshot handcannons.

Edit: specifically, an ingame tag field in the "ItemContainer" attribute of a container within the submarine editor that adds to the "Containable" line, except it would be unique to each container on the submarine.
![Untitled](https://github.com/user-attachments/assets/398480fc-175e-47c3-b74e-2a0dc825edf8)

**Comment by 4-S-H** [comment]
I think the 'items' and 'excludeditems' attributes in the Containable sub-element of ItemContainer are doing what you've described here.

**Reply by Svenn-20** [reply]
You'd be right, but as someone with no modding experience, I'm not confident enough or familiar enough to access and change it outside of what's accessible in the ingame Submarine editor.
Edit: I also don't want to change the whitelist/blacklist for every container of that type, just specific ones on the submarine
